src/db.rs

```diff
@@ -1760,10 +1761,11 @@ mod tests {
         let fp_registry = Arc::new(FailPointRegistry::new());
         let object_store: Arc<dyn ObjectStore> = Arc::new(InMemory::new());
         let path = Path::from("/tmp/test_kv_store");
-        let kv_store = Db::open_with_opts(
+        let kv_store = Db::open_with_fp_registry(
```

Comment:
This is probably always passing because of the single-threaded tokio thing. The test task is never pre-empted between the put for `foo`->`bla` and the subsequent `get`, and so the flush task never runs, so we get the same effect as simulating pausing the wal sst write. I bet if it were a multi-threaded runtime it would sometimes fail.

Reply:
Mm, not following what you're saying here. Why do we care if the flush task runs?

As far as I can tell, the first put in the test writes durably, so it await's until the WAL is flushed to object storage (it's durable). Then we block all future SST writes, so the second put gets stuck in either the mutable or immutable WAL depending on timing. Then we check read committed gives us the first value and read uncommitted gives us the second. 

I ran the test in a while loop with `#[tokio::test(flavor = "multi_thread", worker_threads = 4)]` for 10+ minutes and it hasn't yet failed.

Reply:
Right. I was speculating about why the test has been passing this whole time, even though we didn't pass the failpoint registry in. We want to make sure that the read with level `Committed` only reads data that was flushed by the wal thread. The test tries to ensure this using the failpoint and verifying that the second write ("foo"->"bla") is not reflected by the get. My theory was that even without the failpoint the flush task was always not scheduled before the call to `get`, because the runtime is single-threaded and the main test task is not pre-empted between the put for "foo"->"bla" and the subsequent get.

> I ran the test in a while loop with #[tokio::test(flavor = "multi_thread", worker_threads = 4)] for 10+ minutes and it hasn't yet failed.

Even without the change to pass the registry in? I guess it could just be a timing thing then.

Reply:
Ahh, OK, I see what you're saying. Yes, if I switch back to no FP registry and enable threads, I can get it to fail. I did have to add a 100ms sleep after the put, but it does fail. In any case, now it works properly. :)